modules/site-health/audit-database/class-perflabdbmetrics.php

```diff
@@ -0,0 +1,433 @@
+<?php
+/**
+ * Performance Lab database metrics computation module
+ *
+ * @package performance-lab
+ * @group audit-database
+ *
+ * @since 1.4.0
+ */
+
+/**
+ * Performance Lab database metrics class,
+ *
+ * @package performance-lab
+ * @group audit-database
+ *
+ * @since 1.4.0
+ */
+class PerflabDbMetrics {
+	/** First eligible db version, the advent of utfmb4 in WordPress databases.
+	 *
+	 * @var int
+	 */
+	public $first_compatible_db_version = 32814;
+	/** Latest eligible db version.
+	 * TODO: retest when new DB versions become available.
+	 *
+	 * @var int
+	 */
+	public $last_compatible_db_version = 53496;
+	/** OK to gather metrics when $wp_db_version is greater than $last_compatible_db_version.
+	 *
+	 * @var bool false if we should error out on newer version.
+	 */
+	public $reject_newer_db_versions = false;
+	/** This instances has high-resolution time.
+	 *
+	 * @var bool true The hrtime function is available.
+	 */
+	public $has_hr_time = false;
+
+	/** Utility
+	 *
+	 * @var PerflabDbUtilities singleton instance
+	 */
+	private $util;
+
+	/** Memoized table_info result set.
+	 *
+	 * @var array Table descriptions.
+	 */
+	private $table_info_cache = array();
+
+	/** Memoized db_version result set.
+	 *
+	 * @var object Database version object.
+	 */
+	private $db_version;
+
+	/** Cache for get_variable().
+	 *
+	 * @var array Associative array, in which keys are Status or Variable names
+	 */
+	private $variable_cache = array();
+
+	/** Constructor.
+	 */
```

Comment:
Please add `@since` to all methods, including constructor.
```suggestion
	/**
	 * Constructor.
	 *
	 * @since n.e.x.t
	 */
```